src/emulator/gui/include/gui/functions.h

```diff
@@ -30,16 +35,18 @@ enum GenericDialogState {
     CANCEL_STATE
 };
 
-void init(HostState &host);
-void init_background(HostState &host, const std::string &image_path);
-void load_game_background(HostState &host, const std::string &title_id);
-void draw_begin(HostState &host);
+void init(HostState &host, GuiState &gui);
```

Comment:
From the viewpoint of this module, the focus is the gui, so it's better if we put gui as the first parameter. It's ok if you can reorder this, but i recommend you to do so